src/main/scala/com/wavesplatform/history/CheckpointStorage.scala

```diff
@@ -0,0 +1,35 @@
+package com.wavesplatform.history
+
+import org.h2.mvstore.{MVMap, MVStore}
+import scorex.network.{BlockCheckpoint, Checkpoint}
+import scorex.transaction.CheckpointService
+import scorex.utils.LogMVMapBuilder
+
+trait CheckpointStorage {
+  def checkpoint: java.util.Map[Int, (Seq[(Int, Array[Byte])], Array[Byte])]
+
+  def commit(): Unit
+}
+
+class MVStoreCheckpointStorage(db: MVStore) extends CheckpointStorage {
+  val checkpoint: MVMap[Int, (Seq[(Int, Array[Byte])], Array[Byte])] = db.openMap("checkpoint", new LogMVMapBuilder[Int, (Seq[(Int, Array[Byte])], Array[Byte])])
+
+  def commit(): Unit = db.commit()
+}
+
+class CheckpointServiceImpl(storage: CheckpointStorage) extends CheckpointService {
```

Comment:
The idea behind the special trait is to separate the way we hande data from the way we actually store it. Unfortunately this didn't work out for testing due to mvstore specifics.  But in case we migrate from mvstore, we'll have to switch Storages without(idealistically) touching *ServiceImpl. I'd prefer to keep this separation.